app/controllers/users_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    @user.save
+    session[:user_id] = @user.id
+    flash.notice = "User #{@user.name} has succesfully signed in"
+    redirect_to root_path
+  end
+
+  def show
+    @user = User.find(params[:id])
```

Comment:
- The show action lists events created by user, i.e. @created_events = current_user.events (screenshot from Odin)
- The show action lists events previoulsy attendented by user using event model scope, i.e. @past_events = current_user.attended_events.previous (screenshot from Odin)
- The show action lists upcoming events previously attended by user using event model scope, i.e. @upcoming_events = current_user.attended_events.upcoming (screenshot from Odin)